plugins/InspectionGadgets/InspectionGadgetsAnalysis/src/com/siyeh/ig/psiutils/BoolUtils.java

```diff
@@ -84,6 +87,31 @@ public static String getNegatedExpressionText(@Nullable PsiExpression condition,
     return getNegatedExpressionText(condition, ParenthesesUtils.NUM_PRECEDENCES, tracker);
   }
 
+  /**
+   * Returns the number of logical operands in the expression.
+   *
+   * @param node The expression
+   * @return the number of logical operands in the expression
+   */
+  public static int getNbOperands(@Nullable PsiExpression condition) {
+    PsiExpression unparenthesizedExpression = PsiUtil.skipParenthesizedExprDown(condition);
+    if (!(unparenthesizedExpression instanceof PsiPolyadicExpression)) {
+      return 1;
+    }
+    PsiPolyadicExpression infixExpression= (PsiPolyadicExpression) unparenthesizedExpression;
+    if (!Arrays.asList(ANDAND, OROR).contains(infixExpression.getOperationTokenType())
```

Comment:
Let't avoid unnecessary allocations here, just compare the type with equals

Reply:
Done